integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/processor/test/DeactivatedCappMPBehaviourOnRestartTestCase.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.OMFactory;
+import org.apache.axiom.om.OMNamespace;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.axis2client.AxisServiceClient;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.Utils;
+import org.wso2.esb.integration.common.utils.servers.axis2.SampleAxis2Server;
+
+/**
+ * Testing whether a deactivated Message Processor from a capp is deployed successfully upon ESB server restart.
+ */
+public class DeactivatedCappMPBehaviourOnRestartTestCase extends ESBIntegrationTest {
+    private ServerConfigurationManager serverConfigurationManager;
+    private SampleAxis2Server axis2Server;
+    private static String carFileName = "CappMPServerRestartTestCApp_1.0.0.car";
+    private static final String PROXY_SERVICE_NAME = "CappMPServerRestartTestProxy";
+    private static final String EXPECTED_ERROR_MESSAGE = "Error occurred while deploying Carbon Application";
+    private static final String SUCCESS_MESSAGE = "Deploying Carbon Application : CappMPServerRestartTestCApp_1.0.0.car";
+    private LogViewerClient logViewerClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        context = new AutomationContext("ESB", TestUserMode.SUPER_TENANT_ADMIN);
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        axis2Server = new SampleAxis2Server("test_axis2_server_9001.xml");
+        axis2Server.deployService(SampleAxis2Server.SIMPLE_STOCK_QUOTE_SERVICE);
+        axis2Server.start();
+        uploadCapp(carFileName, new DataHandler(
+                new URL("file:" + File.separator + File.separator + getESBResourceLocation() + File.separator + "car"
+                        + File.separator + carFileName)));
+        isProxyDeployed(PROXY_SERVICE_NAME);
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Testing whether a deactivated Message Processor from a capp is deployed successfully upon ESB server restart")
+    public void testDeactivatedMPUponServerRestart() throws Exception {
+
+        // Stopping the axis2 Server before sending the client request.
+        axis2Server.stop();
+        AxisServiceClient client = new AxisServiceClient();
+        client.sendRobust(createPlaceOrderRequest(3.141593E0, 4, "IBM"), getProxyServiceURLHttp(PROXY_SERVICE_NAME),
+                "placeOrder");
+
+        // Wait till the MP deactivates successfully. It usually takes 2 * 4
```

Comment:
What does 2*4 mean, how does it relate to the thread sleep that's there. Also, rather than making it a static thread sleep, is there a way we can check for it status perioically until it get deactivated?

Reply:
fixed.